src/configuration.rs

```diff
@@ -0,0 +1,588 @@
+    policy: Vec<RawPolicy>,
+    selinux: Option<Selinux>,
+    servicedir: Vec<PathBufElement>,
+    servicehelper: Option<PathBuf>,
+    standard_session_servicedirs: Option<()>,
+    standard_system_servicedirs: Option<()>,
+    syslog: Option<()>,
+    r#type: Vec<TypeElement>,
+    user: Vec<UserElement>,
+}
+impl FromStr for RawConfiguration {
+    type Err = quick_xml::DeError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // TODO: validate expected DOCTYPE
+        // TODO: validate expected root element (busconfig)
+        quick_xml::de::from_str(s)
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+struct RawPolicy {
+    #[serde(rename = "@at_console")]
+    at_console: Option<bool>,
+    #[serde(rename = "@context")]
+    context: Option<PolicyContext>,
+    #[serde(rename = "@group")]
+    group: Option<Principal>,
+    #[serde(default, rename = "$value")]
+    rules: Vec<Rule>,
+    #[serde(rename = "@user")]
+    user: Option<Principal>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+enum Rule {
+    Allow(RuleAttributes),
+    Deny(RuleAttributes),
+}
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
+#[serde(default, rename_all = "lowercase")]
+struct RuleAttributes {
+    #[serde(rename = "@send_interface")]
+    send_interface: Option<RuleMatch>,
+    #[serde(rename = "@send_member")]
+    send_member: Option<RuleMatch>,
+    #[serde(rename = "@send_error")]
+    send_error: Option<RuleMatch>,
+    #[serde(rename = "@send_broadcast")]
+    send_broadcast: Option<bool>,
+    #[serde(rename = "@send_destination")]
+    send_destination: Option<RuleMatch>,
+    #[serde(rename = "@send_destination_prefix")]
+    send_destination_prefix: Option<String>,
+    #[serde(rename = "@send_type")]
+    send_type: Option<RuleMatchType>,
+    #[serde(rename = "@send_path")]
+    send_path: Option<RuleMatch>,
+    #[serde(rename = "@receive_interface")]
+    receive_interface: Option<RuleMatch>,
+    #[serde(rename = "@receive_member")]
+    receive_member: Option<RuleMatch>,
+    #[serde(rename = "@receive_error")]
+    receive_error: Option<RuleMatch>,
+    #[serde(rename = "@receive_sender")]
+    receive_sender: Option<RuleMatch>,
+    #[serde(rename = "@receive_type")]
+    receive_type: Option<RuleMatchType>,
+    #[serde(rename = "@receive_path")]
+    receive_path: Option<RuleMatch>,
+    #[serde(rename = "@send_requested_reply")]
+    send_requested_reply: Option<bool>,
+    #[serde(rename = "@receive_requested_reply")]
+    receive_requested_reply: Option<bool>,
+    #[serde(rename = "@eavesdrop")]
+    eavesdrop: Option<bool>,
+    #[serde(rename = "@own")]
+    own: Option<RuleMatch>,
+    #[serde(rename = "@own_prefix")]
+    own_prefix: Option<String>,
+    #[serde(rename = "@user")]
+    user: Option<RuleMatch>,
+    #[serde(rename = "@group")]
+    group: Option<RuleMatch>,
+}
```

Comment:
It's probably worth applying https://github.com/dbus2/busd/pull/146/files#r1789045200 here, too: encode distinct rule possibilities into an enum